### PR TITLE
[Feature] Add sub_bitmap function

### PR DESCRIPTION
**Pull request by tomscut**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Problem Summary(Required) ：
Add function `sub-bitmap`.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wangsimo0** [comment]
Hi tomscut, 
Thanks for contributing!
Could you please add the user document so we can get better understanding of this function? 
The folder would be starrocks/docs/sql-reference/sql-functions/bitmap-functions/

Thanks again!

**Comment by tomscut** [comment]
> Hi tomscut, Thanks for contributing! Could you please add the user document so users can get better understanding of this function? The folder would be starrocks/docs/sql-reference/sql-functions/bitmap-functions/
> 
> Thanks again!

Thanks @wangsimo0 for your suggestion. I will add the document later.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by tomscut** [comment]
Hi @kangkaisen @stdpain , could you please help review this PR. Thank you.

**Review by kangkaisen — DISMISSED** [review]

**Comment by tomscut** [comment]
Hi @imay @wangsimo0 , could you also please take a look? Thanks.

**Review by stdpain — DISMISSED** [review]

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by kangkaisen — APPROVED** [review]

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by tomscut** [comment]
Hi @stdpain , I rebased the code. Please take a look. Thank you!

File: be/test/exprs/vectorized/bitmap_functions_test.cpp [hunk 1]
```diff
@@ -1944,4 +1944,216 @@ TEST_F(VecBitmapFunctionsTest, array_to_bitmap_test) {
...
+    BitmapValue bitmap({1, 2, 3, 4, 5, 64, 128, 256, 512, 1024});
+    auto bitmap_column = BitmapColumn::create();
+    bitmap_column->append(&bitmap);
+
+    {
+        Columns columns;
+        auto offset = Int64Column::create();
+        auto len = Int64Column::create();
+        offset->append(2);
+        len->append(3);
```
Comment (fzhedu) [inline be/test/exprs/vectorized/bitmap_functions_test.cpp:1957]: what if offset and/or len is null?

File: be/src/exprs/vectorized/bitmap_functions.cpp [hunk 2]
```diff
@@ -483,4 +483,49 @@ ColumnPtr BitmapFunctions::base64_to_bitmap(FunctionContext* context, const star
...
+            len_viewer.value(row) <= 0) {
+            builder.append_null();
+            continue;
+        }
+
+        auto bitmap = bitmap_viewer.value(row);
+        auto offset = offset_viewer.value(row);
+        auto len = len_viewer.value(row);
+        if (bitmap->cardinality() == 0 || offset == INT_MIN || len <= 0) {
+            builder.append_null();
```
Comment (fzhedu) [inline be/src/exprs/vectorized/bitmap_functions.cpp:515]: empty is null?

File: be/test/exprs/vectorized/bitmap_functions_test.cpp (hunk 1 above)
Comment (tomscut) [reply be/test/exprs/vectorized/bitmap_functions_test.cpp:1957]: Thanks @fzhedu for your review. Do you mean to add test cast for `offset and/or len is null`?

File: be/src/exprs/vectorized/bitmap_functions.cpp (hunk 2 above)
Comment (tomscut) [reply be/src/exprs/vectorized/bitmap_functions.cpp:515]: > empty is null?

Yes, the semantics of the current implementation is: empty bitmap will not have sub bitmap. Do you think this needs to be changed? Thanks.

**Review by stdpain — APPROVED** [review]

**Pull request merged**